### PR TITLE
Use ABI names for registers.

### PR DESCRIPTION
**Pull request by timsifive**
This should cause the machine interface register list to list the ABI
names, which in turn will cause them to show up that way in Eclipse.
(Eclipse can apparently not give the registers useful names itself.)

This fixes https://github.com/riscv/riscv-openocd/issues/93

### CODE REVIEW

File: gdb/riscv-tdep.c [hunk 1]
```diff
@@ -75,17 +75,19 @@ struct riscv_frame_cache
...
+/* Use the ABI names here because Eclipse can't display names besides what we
+ * declare here, and users want to see the ABI names in any case. */
 static const char * const riscv_gdb_reg_names[RISCV_LAST_FP_REGNUM + 1] =
 {
-  "x0",  "x1",  "x2",  "x3",  "x4",  "x5",  "x6",  "x7",
-  "x8",  "x9",  "x10", "x11", "x12", "x13", "x14", "x15",
-  "x16", "x17", "x18", "x19", "x20", "x21", "x22", "x23",
-  "x24", "x25", "x26", "x27", "x28", "x29", "x30", "x31",
+  "zero", "ra", "sp", "gp", "tp", "t0", "t1", "t2",
+  "fp", "s1", "a0", "a1", "a2", "a3", "a4", "a5",
```
Comment (ilg-ul) [inline gdb/riscv-tdep.c]: I checked the disassembled code and, at least in my environment, it uses `s0`, not `fp`:

```
40400196:	1141                	addi	sp,sp,-16
40400198:	c622                	sw	s0,12(sp)
4040019a:	0800                	addi	s0,sp,16
```

Comment (aswaterman) [reply gdb/riscv-tdep.c]: Indeed, s0 is the preferred alias for x8.

**Comment by ilg-ul** [comment]
I checked this patch and except the fp/s0 detail, it seems ok, the API register names are shown in the Registers view, see image.
screen shot 2017-11-02 at 10 08 51

**Comment by timsifive** [comment]
Thanks for reviewing.

**Comment by TM1234** [comment]
This doesn't do anything for CSR names.
They're still all CSRn (0 <= n <= 4095) rather than using, say, draft priv abi spec v0.10 names.
So I'm not sure it's correct to say that this mod fixes https://github.com/riscv/riscv-openocd/issues/93.
And the correct solution is still to use the gdb/mi XML mechanism but I guess that nobody has had time yet to understand this fully and propose a solution that uses it (including me!)?

**Comment by ilg-ul** [comment]
I renamed https://github.com/riscv/riscv-openocd/issues/93 to explicitly refer to Core Registers.

**Pull request merged**